src/main/java/org/candlepin/insights/controller/InventoryController.java

```diff
@@ -145,7 +145,10 @@ private void extractHardwareFacts(Map<String, String> pinheadFacts, ConduitFacts
         String memoryTotal = pinheadFacts.get(MEMORY_MEMTOTAL);
         if (!isEmpty(memoryTotal)) {
             try {
-                int memoryBytes = Integer.parseInt(memoryTotal);
+                // Check for special openshift format 1234.00B and parse accordingly.
+                String cleanedMemoryTotal = memoryTotal.replaceAll("\\.\\d*\\D*$", "");
```

Comment:
I'm not sure if this is write.  That regex breaks down to the following:
* a literal "." character
* zero or more numerical characters
* zero or more non-numerical characters

The following are all matches:
* .
* .1
* .B
* .1B

But the string "1234.00B" will not match for example.

Something like "\d+\.\d\dB$" would be a match for the format you've given in the comment, but that regex may not work based on what the rules of that format are (e.g. based on the example it's unclear how long the decimal portion may be, if there is always a decimal portion, etc).

Reply:
What about searching for azAZ at the end of the string and then examining the value? B== Bytes, other values could raise a warning.

Reply:
@awood This works currently because of the use of replaceAll(...) I believe.

How about changing it to do a full text match with: ""\\d*\\.\\d*B"

If matched, strip the B and process as a decimal. I doubt very much that the value will be reported as anything other than this.

If the match fails, it'll get processed as per normal means, and other formats would fail with NumberFormatException and will be skipped as it currently is. This somewhat falls in line with @barnabycourt suggestion. Thoughts?

Reply:
@mstead ah, okay, I was missing the forest for the trees.  You wanted to elide the ".00B" suffix.  In that case, the original regex would have worked.

I like the idea of doing a full match just to make sure we're working with something that's sane and then we could process as a decimal number (or round to an integer if appropriate).

Reply:
:+1:

Reply:
Updated.